apps/test-netzer-encoder/src/Netzer.cpp

```diff
@@ -0,0 +1,57 @@
+#include "Netzer.h"
+
+using namespace Encoder;
+Netzer::Netzer(PinName mosi, PinName miso, PinName sclk, 
+									callback_ptr callback) 
+									: m_spi(mosi, miso, sclk), m_callback(callback){
+	m_spi.format(16,3);
+	m_spi.frequency(DEFAULT_FREQUENCY_HZ);
+	m_spi.set_dma_usage(DMA_USAGE_ALWAYS);
+}
+
+bool Netzer::spi_read(){
+
+	ThisThread::sleep_for(3ms);		// Delay
+
+	// return status
+	int status = m_spi.transfer(tx_buffer, WORDS, rx_buffer, WORDS, event_callback_t(this, &Netzer::Netzer::spi_callback_debug));
+	
+	// status = 0 => SPI transfer started
+  // status = -1 => SPI peripheral is busy
+	return (status == 0);
+}
+
+uint16_t Netzer::get_raw_data(){
+	return m_raw_data;
+}
+
+void Netzer::spi_callback_debug(int events){
+	// upper lower?
+	const uint8_t crcPolynomial = 0x03;
+
+	// Not sure if I'm handling the payload correctly
+	// If I get the raw_data correctly, the rest should work fine (taken from Alex's test encoder)
+  uint16_t raw_data = (static_cast<uint16_t>(rx_buffer[0]) << 8) | (static_cast<uint16_t>(rx_buffer[1]));
```

Comment:
The Netzer DS-25 is quite different from the Broadcom AEAT-6012 here, each transaction is 27 bits (instead of 12 bits in AEAT-6012), so you can't just use 2 bytes to capture all the data. You'll probably need a 4-byte buffer (32 bits)

Reply:
Double check with the datasheet about this

Reply:
oh I see, so I should just capture all the data and turn it into a uint32, then do the same bit manipulation that Alex did?

Reply:
Also, looking at Alex's code, I'm a bit confused as to the `upper = spi.write(0x8000);`, since he ends up shifting it 16 bits. What's the point of that?

Reply:
I think it could be cuz the SPI api can only handle 8-bit or 16-bit data frames, so he's using two 16-bit buffers to grab the full data

Reply:
oh I see thanks